src/main/java/org/jenkinsci/plugins/workflow/cps/CpsStepContext.java

```diff
@@ -215,17 +215,12 @@ public String getDisplayName() {
 
     /**
      * Returns the thread that is executing this step.
-     * Needs to take {@link CpsThreadGroup} as a parameter to prove that the caller is in CpsVmThread.
```

Comment:
Just having a `CpsThreadGroup` is not enough to prove that the caller is on the CPS VM thread as `CpsStepContext.getThreadGroupSynchronously` appears to work fine from any thread.